src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js

```diff
@@ -178,4 +145,241 @@ const IdentifierRoleField = () => {
   );
 };
 
+const PrefillFields = errors => (
+  <Fieldset>
+    <FormRow>
+      <Field
+        name="plugin"
+        label={
+          <FormattedMessage description="Variable prefill plugin label" defaultMessage="Plugin" />
+        }
+        errors={errors.plugin}
+      >
+        <PluginField />
+      </Field>
+    </FormRow>
+
+    <FormRow>
+      <Field
+        name="attribute"
+        label={
+          <FormattedMessage
+            description="Variable prefill attribute label"
+            defaultMessage="Attribute"
+          />
+        }
+        errors={errors.attribute}
+      >
+        <AttributeField />
+      </Field>
+    </FormRow>
+
+    <FormRow>
+      <Field
+        name="identifierRole"
+        label={
+          <FormattedMessage
+            description="Variable prefill identifier role label"
+            defaultMessage="Identifier role"
+          />
+        }
+        errors={errors.identifierRole}
+      >
+        <IdentifierRoleField />
+      </Field>
+    </FormRow>
+  </Fieldset>
+);
+
+const ObjectsAPIPrefillFields = (values, errors) => {
+  const {getFieldProps} = useFormikContext();
+  return (
+    <>
+      <Fieldset>
+        <FormRow>
+          <Field
+            name="plugin"
+            label={
+              <FormattedMessage
+                description="Variable prefill plugin label"
+                defaultMessage="Plugin"
+              />
+            }
+            errors={errors.plugin}
+          >
+            <PluginField />
+          </Field>
+        </FormRow>
+
+        <FormRow>
+          <Field
+            name="attribute"
+            label={
+              <FormattedMessage
+                description="Objects API prefill API group label"
+                defaultMessage="API group"
+              />
+            }
+            errors={errors.apiGroup}
+          >
+            <AttributeField />
+          </Field>
+        </FormRow>
+      </Fieldset>
+
+      <Fieldset
+        title={
+          <FormattedMessage
+            description="Objects API prefill mappings fieldset title"
+            defaultMessage="Mappings"
+          />
+        }
+      >
+        {/* TODO fix formik stuff, remove delete/remove opts, use componentvariables as initialvalues */}
+        <FormRow>
+          <VariableMapping
+            name="mappings"
+            // mapping={values.mappings | []}
+            // onChange={handleChange}
+            // onChange={getFieldProps("mappings").setFieldValue}
+            {...getFieldProps('mappings')}
+            // mapping={[["foo", "bar"], ["bar", "baz"]]}
+            // inputType="select"
```

Comment:
Formik's `FieldArray` is what you're looking for :)

Reply:
I replaced it with the `VariableMapping` defined in `/admin/form_design/logic/actions/dmn/VariableMapping.js`, I did have to make some changes to make it generic and usable for both DMN and prefill attributes, but it basically does everything we need here I think (and I didn't want to copy it :p)